Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+    ) -> DocumentationDataVariants<ParametersSection> {
+        guard let parameters = parameters, !parameters.isEmpty else {
+            guard hasDocumentedReturnValues, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) else {
+                // The symbol has no parameter documentation or return value documentation and none should be synthesized.
+                return DocumentationDataVariants(defaultVariantValue: nil)
+            }
+            // Since the symbol documented its return value synthesize an "error" parameter for it.
+            var variants = DocumentationDataVariants<ParametersSection>()
+            for trait in signatures.keys {
+                variants[trait] = ParametersSection(parameters: []) // Add an empty section so that this language doesn't fallback to the Objective-C content.
+            }
+            variants[.objectiveC] = ParametersSection(parameters: [Parameter(name: "error", contents: Self.objcErrorDescription)]) // This parameter is synthesized and doesn't have a source range.
+            return variants
+        }
+        
+        var variants = DocumentationDataVariants<ParametersSection>()
+        var allKnownParameterNames: Set<String> = []
+        var argumentLabelsAndPossibleParameterNames: [String: Set<String>] = [:]
+        // Group the parameters by name to be able to diagnose parameters documented more than once.
+        let parametersByName = [String: [Parameter]](grouping: parameters, by: \.name)
+        
+        for (trait, signature) in signatures {
+            // Collect all language representations's parameter information from the function signatures.
+            let knownParameterNames = Set(signature.parameters.map(\.name))
+            allKnownParameterNames.formUnion(knownParameterNames)
+            for parameter in signature.parameters {
+                if let externalName = parameter.externalName {
+                    argumentLabelsAndPossibleParameterNames[externalName, default: []].insert(parameter.name)
+                }
+            }
+            
+            // Remove documented parameters that don't apply to this language representation's function signature.
+            var languageApplicableParameters = parametersByName.filter { knownParameterNames.contains($0.key) }
+            guard !languageApplicableParameters.isEmpty else { 
+                // Add an empty section so that this language doesn't fallback to another language's content.
+                variants[trait] = ParametersSection(parameters: [])
+                continue
+            }
+            
+            // Add a missing error parameter documentation if needed.
+            if trait == .objectiveC, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) {
+                languageApplicableParameters["error"] = [Parameter(name: "error", contents: Self.objcErrorDescription)] // This parameter is synthesized and doesn't have a source range.
+            }
```

Comment:
I ran into a bug here while testing on an actual framework... I think you need to move the `shouldAddObjectiveCErrorParameter` check up ahead of the `guard` statement that tests `languageApplicableParameters.isEmpty`. I ran into this when I added invalid parameters to an Obj-C function that returned an error via an error parameter, but had no other parameters. What happens now is that the guard statement thinks there are no parameters, and so the error parameter is not added at all. Worse, DocC Render for some reason runs into trouble and fails to render the page at all, after a JavaScript error.

If you move the call to `shouldAddObjectiveCErrorParameter` up, the code will add the synthesized error parameter first, and then the `guard` statement will not set the parameter section to be empty.

If you're curious and want to try this yourself, here's the page I used as a test case: https://developer.apple.com/documentation/servicemanagement/smappservice/register()?language=objc

```suggestion
            // Remove documented parameters that don't apply to this language representation's function signature.
            var languageApplicableParameters = parametersByName.filter { knownParameterNames.contains($0.key) }
            
            // Add a missing error parameter documentation if needed.
            if trait == .objectiveC, Self.shouldAddObjectiveCErrorParameter(signatures, parameters) {
                languageApplicableParameters["error"] = [Parameter(name: "error", contents: Self.objcErrorDescription)] // This parameter is synthesized and doesn't have a source range.
            }

            guard !languageApplicableParameters.isEmpty else { 
                // Add an empty section so that this language doesn't fallback to another language's content.
                variants[trait] = ParametersSection(parameters: [])
                continue
            }
```

Reply:
Thanks. I will look at this next week.

Reply:
I fixed the issue where DocC Render didn't display anything because the JSON patches got offset when the parameters section was completely empty. Then added a `testFunctionWithOnlyErrorParameter()` test that covers this example in  https://github.com/apple/swift-docc/pull/776/commits/844e530fcc705de9b88a5263831e35cf27997bf1